lib/YourlsProxy.php

```diff
@@ -47,7 +47,7 @@ class YourlsProxy
      */
     public function __construct(Configuration $conf, $link)
     {
-        if (strpos($link, $conf->getKey('basepath') . '?') === false) {
+        if (strpos($link, $conf->getKey('basepath') . '?') !== 0) {
```

Comment:
Such a slight change and IMHO it's not obvious why it fixes the issue. Here PHP is really bad given this position function is not easy to understand and that's probably why this issue has been introduced. Because actually modern PHP (v8+) seems to have the proper very easily understandable solution, which I'd also suggest here:
```suggestion
        if (str_starts_with($link, $conf->getKey('basepath') . '?') === false) {
```

https://www.php.net/manual/de/function.str-starts-with.php

Reply:
Also maybe we should check `strpos` in our code base to find similar errors hmm? Is not this some kind of common mistake with `strpos`  where better alternatives are available?

looking through the net, apparently I am right, it _is_ a common failure:'
* https://core.trac.wordpress.org/ticket/58012
* https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6419 (AFAIK we use this project, but sadly it's only proposed yet)

Reply:
We still support PHP 7.3 & 7.4 and `str_starts_with` requires PHP 8. I can add a comment if you insist, though git blame should already cover the reason for the change.